src/ncp/ncp_base.cpp

```diff
@@ -2710,6 +2710,14 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_PHY_CALIBRATED_POWER>
 }
 #endif // OPENTHREAD_CONFIG_PLATFORM_POWER_CALIBRATION_ENABLE
 
+#if OPENTHREAD_FTD || OPENTHREAD_MTD
+template <> otError NcpBase::HandlePropertyGet<SPINEL_PROP_RADIO_CAPS>(void)
```

Comment:
We already have one of the following emitted from `SPINEL_PROP_CPAS`:

```c++
#if OPENTHREAD_FTD
    SuccessOrExit(error = mEncoder.WriteUintPacked(SPINEL_CAP_CONFIG_FTD));
#elif OPENTHREAD_MTD
    SuccessOrExit(error = mEncoder.WriteUintPacked(SPINEL_CAP_CONFIG_MTD));
#elif OPENTHREAD_RADIO
    SuccessOrExit(error = mEncoder.WriteUintPacked(`));
#endif
```

The first two `SPINEL_CAP_CONFIG_MTD` and `SPINEL_CAP_CONFIG_FTD` indicate it is a NCP, and `SPINEL_CAP_CONFIG_RADIO` indicates an RCP. 

Can we reuse the same? Do we need new `SPINEL_CAP_NCP`?

Reply:
I think the goal is to define a new (NCP-like) architecture. 

It may be helpful to use a different term for it other than "NCP" to help distinguish it from earlier NCP/RCP architectures? 

An idea: Would it support BR functions, if so, may be "BCP" for BR co-processor?

Reply:
Let's directly reuse `SPINEL_CAP_NCP`. We don't need this PR any more. WIll close this. I've implemented the spinel initialization in this [commit](https://github.com/openthread/openthread/pull/9898/commits/5c9e988e6c4de8a3c2ba87e4d35ab15cd4161c37).  

> I think the goal is to define a new (NCP-like) architecture.
> 
> It may be helpful to use a different term for it other than "NCP" to help distinguish it from earlier NCP/RCP architectures?

I think the purpose to distinguish old NCP and new NCP is that we can report an error during initialization if we find the co-processor is an old NCP. Maybe we can do that by adding another SPINEL_PROP_CAP to indicate that? Old NCP won't contain that CAP while new NCP will have the CAP. Otherwise we will add new type 'BCP' as the peer of 'FTD/MTD/RCP' which will make things more complicated. (In OT, there are a lot of code wrapped by FTD/MTD).